### PR TITLE
Implemented Formik & Yup & CustomFields

### PR DESCRIPTION
**Pull request by aliyorulmazdev**

### CODE REVIEW

File: prExplorer/client-app/src/app/common/form/CustomSelectInput.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { useField } from 'formik';
+import React from 'react';
+import { Form, Label, Select } from 'semantic-ui-react';
+
+interface CustomSelectInputProps {
```
Comment (mm-dsibinski) [inline prExplorer/client-app/src/app/common/form/CustomSelectInput.tsx:5]: Thanks for proper naming! :)

File: prExplorer/client-app/src/app/common/form/CustomSelectInput.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    options: any;
+    label?: string;
+}
+
+export default function CustomSelectInput(customSelectInputProps: CustomSelectInputProps) {
+    const [field, meta, helpers] = useField(customSelectInputProps.name);
+    return (
+        <Form.Field error={meta.touched && !!meta.error}>
+            <label>{customSelectInputProps.label}</label>
+            <Select clearable options={customSelectInputProps.options} value={field.value || null} onChange={(e, d) => helpers.setValue(d.value)}
```
Comment (mm-dsibinski) [inline prExplorer/client-app/src/app/common/form/CustomSelectInput.tsx:17]: @aliyorulmazdev are you sure you're using prettier in VS Code? Your code does not seem to be formatted

File: prExplorer/client-app/src/app/common/form/CustomSelectInput.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    placeholder: string;
+    name: string;
+    options: any;
+    label?: string;
+}
+
+export default function CustomSelectInput(customSelectInputProps: CustomSelectInputProps) {
+    const [field, meta, helpers] = useField(customSelectInputProps.name);
+    return (
+        <Form.Field error={meta.touched && !!meta.error}>
```
Comment (dsibinski) [inline prExplorer/client-app/src/app/common/form/CustomSelectInput.tsx:15]: @aliyorulmazdev are u sure you use Prettier in VS Code? Your code does not seem to be formatted

File: prExplorer/client-app/src/app/common/form/CustomTextArea.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import { useField } from 'formik';
+import React from 'react';
+import { Form, Label } from 'semantic-ui-react';
+
+interface CustomTextAreaProps {
+    placeholder: string;
+    name: string;
+    rows: number;
```
Comment (dsibinski) [inline prExplorer/client-app/src/app/common/form/CustomTextArea.tsx:8]: `rows` seems to not be used

File: prExplorer/client-app/src/app/common/form/CustomTextArea.tsx [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    label?: string;
+}
+
+export default function CustomTextInput(customTextAreaProps: CustomTextAreaProps) {
+    const [field, meta] = useField(customTextAreaProps.name);
+    return (
+        <Form.Field error={meta.touched && !!meta.error}>
+            <label>{customTextAreaProps.label}</label>
+            <textarea {...field} {...customTextAreaProps} />
+            {meta.touched && meta.error ? (
```
Comment (dsibinski) [inline prExplorer/client-app/src/app/common/form/CustomTextArea.tsx:18]: It seems that this part for displaying validation errors could be extracted as a separate component. It depends on the type of `meta`, but I guess you can make it work

File: prExplorer/client-app/src/app/stores/productStore.ts [hunk 6]
```diff
@@ -114,6 +117,7 @@ export default class ProductStore {
       runInAction(() => {
         this.productRegistry.delete(id);
         this.loading = false;
+        toast.success("Succesfully Deleted Product");
```
Comment (dsibinski) [inline prExplorer/client-app/src/app/stores/productStore.ts:120]: store isn't the best place for displaying toasts. See that the store is purely related to data processing/business logic, while `toast` is something that depends on UI. 
In other words, if someone calls you store's method from a non-UI context, displaying `toast` would not be desired or may even produce some errors

**Review by dsibinski — APPROVED** [review]
@aliyorulmazdev I left few remarks, you can implement them now or later, up to you.
In general, it looks good! It's very good practice that you use `Yup` and I even have another idea how you can make use of it :) I'll add a task for that

File: prExplorer/client-app/src/app/common/form/CustomTextArea.tsx (hunk 4 above)
Comment (aliyorulmazdev) [reply prExplorer/client-app/src/app/common/form/CustomTextArea.tsx:8]: I use it on ProductForm.tsx 
            <CustomTextArea
              rows={3}
              placeholder="Description"
              name="description"
            />

File: prExplorer/client-app/src/app/common/form/CustomSelectInput.tsx (hunk 3 above)
Comment (aliyorulmazdev) [reply prExplorer/client-app/src/app/common/form/CustomSelectInput.tsx:15]: I will check that

File: prExplorer/client-app/src/app/stores/productStore.ts (hunk 6 above)
Comment (aliyorulmazdev) [reply prExplorer/client-app/src/app/stores/productStore.ts:120]: so i can use it on my ProductForm.tsx handleFormSubmit function.

**Pull request merged**